### PR TITLE
When-a-app-is-hidden,-unhidden,-added-or-removed,-do-not-refresh-the-…

### PR DESCRIPTION
**Pull request by Perumal605**
…page

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208361834831577

### CODE REVIEW

File: routes/launcher.php [hunk 1]
```diff
@@ -18,23 +18,24 @@
...
 
-$r->group('/apps/launcher/{key}', function ( RouteCollectionInterface $r ) {
+$r->group( '/apps/launcher/{key}', function ( RouteCollectionInterface $r ) {
     $r->get( '/app/{slug}', [ AppController::class, 'show' ] );
     $r->get( '/', [ LauncherController::class, 'show' ] );
     $r->get( '/training', [ TrainingController::class, 'show' ] );
     $r->get( '/logout', [ LoginController::class, 'logout' ] );
-})->middleware( new LoggedIn() )
-    ->middleware( new CheckShareCookie() );
+    $r->get( '/get-apps', [ AppController::class, 'get_apps' ] );
```
Comment (incraigulous) [inline routes/launcher.php:26]: Change this to `$r->get( '/apps', [ AppController::class, 'all' ] );`

File: src/Controllers/MagicLink/AppController.php [hunk 2]
```diff
@@ -267,4 +234,41 @@ public function reset_apps( Request $request )
...
+    public function get_apps( Request $request ) {
+        // Fetch all apps
+        $user       = get_current_user_id();
+        $apps       = container()->get( Apps::class );
+        $apps_array = $apps->for_user( $user );
+
+        return response( $apps_array );
+    }//end get_apps()
+
+/**
```
Comment (incraigulous) [inline src/Controllers/MagicLink/AppController.php:247]: Your linter seems to be removing the indentation on the first line of comments for some reason. Please make sure the only linter you are using is PHPCS.

File: src/Controllers/MagicLink/AppController.php [hunk 3]
```diff
@@ -267,4 +234,41 @@ public function reset_apps( Request $request )
 
         return response( [ 'message' => 'App order updated' ] );
     }//end reset_apps()
+
+    public function get_apps( Request $request ) {
```
Comment (incraigulous) [inline src/Controllers/MagicLink/AppController.php:238]: You don't seem to be using the request parameter, so you are safe to omit it.

**Review by incraigulous — CHANGES_REQUESTED** [review]
Your formatter seems to removing the indentation on the first line of comments for some reason. 

Screenshot 2024-09-25 103155

File: src/Controllers/MagicLink/AppController.php (hunk 3 above)
Comment (Perumal605) [reply src/Controllers/MagicLink/AppController.php]: @incraigulous  "The change request has been updated."

**Review by incraigulous — APPROVED** [review]

**Pull request merged**